awx/api/views/root.py

```diff
@@ -350,22 +351,25 @@ def post(self, request):
                         extra=dict(actor=request.user.username))
             return Response({"error": _("Invalid JSON")}, status=status.HTTP_400_BAD_REQUEST)
 
-        # Save Entitlement Cert/Key
+        from awx.main.utils.common import get_licenser
         license_data = json.loads(data_actual)
-        if 'entitlement_cert' in license_data:
-            settings.ENTITLEMENT_CERT = license_data['entitlement_cert']
+        if 'manifest' in license_data:
+            try:
+                license_data = validate_entitlement_manifest(license_data['manifest'])
+            except Exception as e:
+                logger.exception('Invalid license submitted.')
+                return Response({"error": 'Invalid license submitted.'}, status=status.HTTP_400_BAD_REQUEST)
 
-        try:
-            # Validate entitlement cert and get subscription metadata
-            # validate() will clear the entitlement cert if not valid
-            from awx.main.utils.common import get_licenser
-            license_data_validated = get_licenser().validate(new_cert=True)
-        except Exception:
-            logger.warning(smart_text(u"Invalid license submitted."),
-                           extra=dict(actor=request.user.username))
-            # If License invalid, clear entitlment cert value
-            settings.ENTITLEMENT_CERT = ''
-            return Response({"error": _("Invalid License")}, status=status.HTTP_400_BAD_REQUEST)
+            try:
+                # Validate entitlement cert and get subscription metadata
```

Comment:
super picky: comments don't reflect what we're now doing

Reply:
correct, this PR needs a lot of love and nitpickiness; for now we're settling on "does it work"

Reply:
I'd like to take the *entire* PR, squash it, and aggressively clean up cruft and write tests.